tensorflow/core/kernels/ragged_tensor_to_tensor_op.cc

```diff
@@ -166,19 +169,25 @@ class RaggedTensorToTensorBaseOp : public OpKernel {
     result->reserve(output_shape_proto.dim_size());
     for (const TensorShapeProto::Dim& dim : output_shape_proto.dim()) {
       // Note that this may be -1 (if dimension size is unknown).
-      result->push_back(dim.size());
+        INDEX_TYPE dim_size = dim.size();
+        if (dim_size < 0) {
+            if (dim_size == -1) {
+                dim_size = (&dim == &output_shape_proto.dim(0)) ? first_dim : 0;
+            } else {
+                return errors::InvalidArgument(
+                    "Invalid dimension size: ", dim_size);
+            }
+        }
+        result->push_back(dim_size);
     }
 
-    if ((*result)[0] < 0) {
-      (*result)[0] = first_dim;
-    }
     for (int i = 1; i <= ragged_rank_; ++i) {
-      if ((*result)[i] < 0) {
-        TF_RETURN_IF_ERROR(GetMaxWidth(c, i, &(*result)[i]));
-      }
+        if ((*result)[i] < 0) {
+            TF_RETURN_IF_ERROR(GetMaxWidth(c, i, &(*result)[i]));
+        }
```

Comment:
```suggestion
      if ((*result)[i] < 0) {
        TF_RETURN_IF_ERROR(GetMaxWidth(c, i, &(*result)[i]));
      }
```

Let's keep the spacing as it exists.